src/components/vr_cooperation/src/command/base_json_request.cc

```diff
@@ -143,6 +144,7 @@ void BaseJsonRequest::SendMessageToMobile(
                      this);
   LOG4CXX_DEBUG(logger_, "Message to Mob: " << message->json_message());
   parent_->SendMessageToMobile(message);
+  parent_->UnregisterRequest(message->correlation_id());
```

Comment:
I'm not sure that unregistering request should be here because this action deletes the command and you never do receive response.